### PR TITLE
feat: use new log group subscription module

### PR DESCRIPTION
**Pull request by luke-zhu**
This change replaces the old cloudwatch logs subscription with the new stuff in https://github.com/observeinc/terraform-aws-cloudwatch-logs-subscription.

This exposes a few new variables and removes the subscribe-to-lambda variable.

Testing: Successfully applied in Thunderdome.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -59,37 +59,49 @@ variable "dead_letter_queue_destination" {
...
+
+    See https://github.com/observeinc/terraform-aws-cloudwatch-logs-subscription#input_log_group_excludes for more info"
+  EOF
   type        = list(string)
   default     = []
 }
 
+variable "subscribed_log_group_filter_pattern" {
+  description = <<-EOF
+    A CloudWatch Subscription filter filter pattern.
```
Comment (jta) [inline variables.tf:90]: filter filter

File: variables.tf [hunk 2]
```diff
@@ -156,3 +168,9 @@ variable "cloudwatch_metrics_exclude_filters" {
   type        = list(string)
   default     = []
 }
+
+variable "cloudtrail_enable" {
```
Comment (jta) [inline variables.tf:172]: Move this to a separate PR.

File: variables.tf [hunk 3]
```diff
@@ -59,37 +59,49 @@ variable "dead_letter_queue_destination" {
   description = "Send failed events/function executions to a dead letter queue arn sns or sqs"
 }
 
-variable "subscribed_log_group_names" {
```
Comment (jta) [inline variables.tf:62]: Removing this would trigger a backward incompatible change. Is there any way of supporting both `subscribed_log_group_names` and `subscribed_log_group_matches`, but making them mutually exclusive?

File: variables.tf [hunk 4]
```diff
@@ -59,37 +59,49 @@ variable "dead_letter_queue_destination" {
...
+variable "subscribed_log_group_matches" {
+  description = <<-EOF
+    A list of regex patterns describing CloudWatch log groups to subscribe to.
 
-variable "subscribed_s3_bucket_arns" {
-  description = "List of additional S3 bucket ARNs to subscribe lambda to."
+    See https://github.com/observeinc/terraform-aws-cloudwatch-logs-subscription#input_log_group_matches for more info"
+  EOF
+  type        = list(string)
+  default     = [".*"]
```
Comment (jta) [inline variables.tf:75]: Might be a brave default for now.

My suggestion is to start this off as an empty list and retain backward compatibility with `subscribed_log_group_names`. We can update examples to include `".*"` and trial it on a few different target accounts.

Once we're confident that this works robustly across many different environments, we can flip the default and increment the major version of the module, providing upgrading instructions and removing `subscribed_log_group_names`.

**Review by jta — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (luke-zhu) [reply variables.tf:90]: done

File: variables.tf (hunk 4 above)
Comment (luke-zhu) [reply variables.tf:75]: done

File: firehose.tf [hunk 5]
```diff
@@ -1,4 +1,4 @@
-resource "aws_cloudwatch_log_group" "group" {
+resource "aws_cloudwatch_log_group" "firehose" {
```
Comment (jta) [inline firehose.tf:1]: Changing the attribute name will cause the log group to be recreated.  Not sure it's worth the risk given the attribute name is not exposed externally.

**Review by jta — CHANGES_REQUESTED** [review]

File: firehose.tf (hunk 5 above)
Comment (luke-zhu) [reply firehose.tf:1]: done

**Review by jta — APPROVED** [review]
